cdap-ui/app/features/tracker/controllers/audit-ctrl.js

```diff
@@ -83,18 +88,111 @@ class TrackerAuditController {
     this.myTrackerApi.getAuditLogs(params)
       .$promise
       .then((response) => {
+        this.isTrackerSetup = true;
         this.auditLogs = response;
       }, (err) => {
-        console.log('Error: ', err);
-        this.myAlertOnValium.show({
-          type: 'danger',
-          content: 'Enable the Tracker extension to view audit logs.'
-        });
+        if (err.statusCode === 503) {
+          this.isTrackerSetup = false;
+        } else {
+          this.myAlertOnValium.show({
+            type: 'danger',
+            content: err.data
+          });
+        }
       });
   }
+
+  enableTracker() {
+    this.enableTrackerLoading = true;
+    this.myTrackerApi.getTrackerApp({
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    })
+    .$promise
+    .then(() => {
+      // start programs
+      this.startPrograms();
+    }, () => {
+      // create app
+      this.createTrackerApp();
+    });
+  }
+
+  createTrackerApp() {
+    this.myTrackerApi.getCDAPConfig({ scope: this.$scope })
+      .$promise
+      .then((res) => {
+        let zookeeper = res.filter( (c) => {
```

Comment:
Would there be a case where the filter returned `[]` array? if thats the case this will be a JS error as `[][0]` will be `undefined` and `undefined.value` will be an error.,